src/components/search/search.component.jsx

```diff
@@ -39,63 +74,98 @@ const handleClick = props.handleClick;
     dispatch(updateCounty(""));
   };
 
+  // Update county search
   const updateCountySearch = (e) => {
     dispatch(updateCounty(e.target.value));
-   
+    setCountiesList(
+      counties_list.filter((c) => {
+        return c.toLowerCase().includes(e.target.value.toLowerCase());
+      })
+    );
   };
 
+  // Update city search
   const updateCitySearch = (e) => {
     dispatch(updatCity(e.target.value));
-  }
 
+    counties.map((c) => {
+      if (Object.keys(c)[0] === county) {
```

Comment:
can you please explain the logic here?

Reply:
This code updates the list of cities displayed based on user input for a specific county. It filters the cities to show only those that match the search criteria provided  by the user.

https://github.com/peviitor-ro/search-engine/assets/67306273/560a52ad-36d3-4062-9930-1c2bf2b80a0c